engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -964,7 +965,11 @@ else if (allocatedContainers.isEmpty() && numRequestedContainers == 0 && dnmgr.c
         appDone = true;
       }
 
-      LOG.debug("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" + numTotalContainers + ", requested=" + numRequestedContainers + ", released=" + numReleasedContainers + ", completed=" + numCompletedContainers + ", failed=" + numFailedContainers + ", currentAllocated=" + allocatedContainers.size());
+      if (!StringUtils.isEmpty(dnmgr.shutdownDiagnosticsMessage)) {
```

Comment:
Here the log message should say along the lines that "Shutdown was requested, but not honored because of the following reason."